### ISSUE TITLE
Swift 3 version Xcode 8.3.1

### ISSUE THREAD

**Issue by kmatskevich** [brave/browser-ios#799]
Hello guys

Did anyone successfully converted the current source to the swift3?

**Comment by jhreis** [comment]
No, we have not started that process yet. See 1.4 for latest, going to merge this into master later today.

Swift 3 conversion will be one of our next biggest items to address. We will need to update all of our dependencies as well, so it will be no small task.

**Comment by kunalverma25** [comment]
@jhreis 
Any timeframe for this task?

**Comment by jhreis** [comment]
No specific timeframe, but you are all free to start the process 😄 

One of the largest pieces will be handling dependency deprecations and upgrades. We share many dependencies as Firefox iOS, so that would be a decent place to look at for migrating some of the extensions and 3rd party code.

I will create a Swift3 branch, to handle any PRs, and for people to pull in changes from other contributors.

**Comment by jhreis** [comment]
We have started the swift migration. Please see the [Swift3](https://github.com/brave/browser-ios/tree/swift3) branch to help out 😄 

Feel free to inquire with any questions.
